sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -521,38 +522,99 @@ class Analyzer(
    */
   object ResolveSortReferences extends Rule[LogicalPlan] {
     def apply(plan: LogicalPlan): LogicalPlan = plan resolveOperators {
-      case s @ Sort(ordering, global, p @ Project(projectList, child))
-          if !s.resolved && p.resolved =>
-        val (newOrdering, missing) = resolveAndFindMissing(ordering, p, child)
+      case s @ Sort(_, _, a: Aggregate) if a.resolved =>
```

Comment:
Unfortunately, it is not trivial. : ( So far, the current rule `ResolveSortReferences` only can handle the missing attributes. In this case, we have to push the aggregate function down to the underlying `Aggregate`. Thus, it does not work.